samples/switchless/README.md

```diff
@@ -0,0 +1,184 @@
+# Switchless Calls Sample
+
+This sample demonstrates how to make switchless calls to host from inside an enclave.
+It has the following properties:
+
+- Explain the concept of switchless calls
+- Identify cases where switchless calls are appropriate
+- Demonstrate how to mark a function as `transition_using_threads` in EDL, and use [`oeedger8r`](https://github.com/openenclave/openenclave/tree/master/docs/GettingStartedDocs/Edger8rGettingStarted.md) tool to compile it
+- Demonstrate how to configure an enclave to enable switchless calls originated within it
+- Recommend the number of host worker threads required for switchless calls in practice
+
+Prerequisite: you may want to read [Common Sample Information](../README.md#common-sample-information) before going further.
+
+## Switchless Calls
+
+In an enclave application, the host makes **ECALL**s into functions exposed by the enclaves it created. Likewise,
+the enclaves may make **OCALL**s into functions exposed by the host that created them. In either case, the
+execution has to be transitioned from an untrusted environment to a trusted environment, or vice versa. Since the
+transition is costly due to heavy security checks, it might be more performance advantageous to make the calls
+**context-switchless**: the caller delegates the function call to a worker thread in the other environment, which does the real job of calling the function and post the result to the caller. Both the calling thread and the
+worker thread never leave their respective execution contexts during the perceived function call.
+
+The calling thread and the worker thread need to exchange information twice during the call. When the switchless
+call is initiated, the caller needs to pass the `job` (encapsulating information regarding the function call in a
+ single object, for details see the next section) to the worker thread. And when the call finishes, the worker
+thread needs to pass the result back to the caller. Both exchanges need to be synchronized.
+
+While switchless calls save transition time, they require at least one additional thread to service the calls.
+More threads typically means more competition of the CPU cores and more context switches, hurting the performance.
+Whether to make a particular function switchless has to weigh the associated costs and savings. In general, the
+good candidates for switchless calls are functions that are: 1) short, thus the transition takes relatively high
+percentage of the overall execution time of the call; and 2) called frequently, so the savings in transition time
+add up.
+
+## How does OpenEnclave support switchless OCALLs
```

Comment:
"Open Enclave" (with a space).  Same in line 37, 38, etc.

Reply:
Fixed.